packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp

```diff
@@ -78,7013 +73,9090 @@
+      if(this->val[i] > other.val[i]) {
+        return true;
+      }
+      // if next value is smaller return false;
+      else { // (this->val[i] > other.val[i])
+        return false;
+      }
     }
-};// uSortItem;
+    // if they are totally equal.
+    return this->index > other.index;
+  }
+};
 
 /*! \brief Sort items for quick sort function.
- *
  */
 template <class IT, class WT>
 struct uSortItem
 {
-    IT id;
-    //unsigned int val;
-    WT val;
-};// uSortItem;
+  IT id;
+  WT val;
+};
 
 /*! \brief Quick sort function.
- *      Sorts the arr of uSortItems, with respect to increasing vals.
+ *  Sorts the arr of uSortItems, with respect to increasing vals.
+ *  TODO: Document input params
  */
 template <class IT, class WT>
-void uqsort(IT n, uSortItem<IT, WT> * arr)
-{
-
-    int NSTACK = 50;
-    int M = 7;
-    IT         i, ir=n, j, k, l=1;
-    IT         jstack=0, istack[50];
-    WT aval;
-    uSortItem<IT,WT>    a, temp;
-
-    --arr;
-    for (;;)
-    {
-        if (ir-l < M)
-        {
-            for (j=l+1;j<=ir;j++)
-            {
-                a=arr[j];
-                aval = a.val;
-                for (i=j-1;i>=1;i--)
-                {
-                    if (arr[i].val <= aval)
-                        break;
-                    arr[i+1] = arr[i];
-                }
-                arr[i+1]=a;
-            }
-            if (jstack == 0)
-                break;
-            ir=istack[jstack--];
-            l=istack[jstack--];
-        }
-        else
-        {
-            k=(l+ir) >> 1;
-
-            ZOLTAN2_ALGMULTIJAGGED_SWAP(arr[k],arr[l+1], temp)
-            if (arr[l+1].val > arr[ir].val)
-            {
-                ZOLTAN2_ALGMULTIJAGGED_SWAP(arr[l+1],arr[ir],temp)
-            }
-            if (arr[l].val > arr[ir].val)
-            {
-                ZOLTAN2_ALGMULTIJAGGED_SWAP(arr[l],arr[ir],temp)
-            }
-            if (arr[l+1].val > arr[l].val)
-            {
-                ZOLTAN2_ALGMULTIJAGGED_SWAP(arr[l+1],arr[l],temp)
-            }
-            i=l+1;
-            j=ir;
-            a=arr[l];
-            aval = a.val;
-            for (;;)
-            {
-                do i++; while (arr[i].val < aval);
-                do j--; while (arr[j].val > aval);
-                if (j < i) break;
-                ZOLTAN2_ALGMULTIJAGGED_SWAP(arr[i],arr[j],temp);
-            }
-            arr[l]=arr[j];
-            arr[j]=a;
-            jstack += 2;
-            if (jstack > NSTACK){
-                std::cout << "uqsort: NSTACK too small in sort." << std::endl;
-                exit(1);
-            }
-            if (ir-i+1 >= j-l)
-            {
-                istack[jstack]=ir;
-                istack[jstack-1]=i;
-                ir=j-1;
-            }
-            else
-            {
-                istack[jstack]=j-1;
-                istack[jstack-1]=l;
-                l=i;
-            }
+void uqsort(IT n, uSortItem<IT, WT> * arr) {
```

Comment:
You don't need this.  If you define `operator<` on `uSortItem`, then you can use `std::sort`.

Reply:
@kddevin Recommended changes would simplify code a lot. But this code is not really changed with the PR, just cleaned up formatting, so we may want to consider as a separate PR.

Reply:
This is a function that almost certainly will never get tested, and thus introduces technical debt on creation.  I'm not your boss, but I think now is a good time to clean it up.

Reply:
I removed operator> and logic now uses just operator< but did not switch to the std::sort - not sure exactly what uqsort does.

Reply:
@MicheldeMessieres wrote:
> not sure exactly what uqsort does

It looks like a non-recursive variant of quicksort.  More untested functions, joy ;-)

Reply:
@kddevin uqsort is currently only tested through migration, which won't execute in our current test suite. So we have a bunch of methods related to migration not default tested. I think adding these should be straight forward. We should probably have some tests to use the Zoltan migration option as well.

Reply:
I'm not your boss, but it seems like adding unit tests for yet another sort routine is a waste of time, vs. just making it work with `std::sort`.

Reply:
@MicheldeMessieres You are correct that we could use a migration test in the test suite.  It isn't required for this PR.  Perhaps we can create a separate issue so that we don't forget to do it.  My concern is that the test might be large for standard Trilinos testing; I'd have to look at the details of migration for that.